compaction/compaction_manager.hh

```diff
@@ -488,10 +488,11 @@ protected:
     future<stop_iteration> maybe_retry(std::exception_ptr err, bool throw_on_abort = false);
 
     // Compacts set of SSTables according to the descriptor.
-    using release_exhausted_func_t = std::function<void(const std::vector<sstables::shared_sstable>& exhausted_sstables)>;
-    future<sstables::compaction_result> compact_sstables_and_update_history(sstables::compaction_descriptor descriptor, sstables::compaction_data& cdata, release_exhausted_func_t release_exhausted,
+    // called when a compaction replaces the exhausted sstables with the new set
+    using on_replace_func_t = std::function<void(const sstables::compaction_completion_desc& desc)>;
+    future<sstables::compaction_result> compact_sstables_and_update_history(sstables::compaction_descriptor descriptor, sstables::compaction_data& cdata, on_replace_func_t replace_exhausted,
                                 compaction_manager::can_purge_tombstones can_purge = compaction_manager::can_purge_tombstones::yes);
-    future<sstables::compaction_result> compact_sstables(sstables::compaction_descriptor descriptor, sstables::compaction_data& cdata, release_exhausted_func_t release_exhausted,
+    future<sstables::compaction_result> compact_sstables(sstables::compaction_descriptor descriptor, sstables::compaction_data& cdata, on_replace_func_t replace_exhausted,
                                 compaction_manager::can_purge_tombstones can_purge = compaction_manager::can_purge_tombstones::yes);
     future<> update_history(::compaction::table_state& t, const sstables::compaction_result& res, const sstables::compaction_data& cdata);
     bool should_update_history(sstables::compaction_type ct) {
```

Comment:
By the way, we also need to patch compaction class to have a random UUID for gc-only sstables. It's simple and will complete the fix, by making sure that gc-only sstables won't replace the regular output sstables in the output sstable run.

Reply:
yeah. i knew. was not sure where i should assign this run-id. should be fixed now.